core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -279,6 +282,10 @@ public RelNode getRoot() {
     return ImmutableList.copyOf(materializations);
   }
 
+  @Override public void registerMaterializationRules(List<UnifyRule> rules) {
+    materializationRules = ImmutableList.copyOf(rules);
+  }
+
```

Comment:
This interface is a little confusing. Because even user does not call this method, normal rules are still be added. Maybe `addExtraMaterializationRules`?

Reply:
ok,make sense.